apps/sms/test/marionette/lib/views/conversation/view.js

```diff
@@ -43,6 +81,86 @@ ConversationView.prototype = {
     return this._switchToContactsPageObject();
   },
 
+  openParticipants: function() {
+    this.accessors.headerTitle.tap();
+
+    var ParticipantsView = require('../participants/view');
+    var participantsView = new ParticipantsView(this.client);
+    participantsView.accessors.waitToAppear();
+    return participantsView;
+  },
+
+  openReport: function(messageId) {
+    var message = this.accessors.findMessage(messageId);
+
+    // Show context menu on message bubble. "1" is the waiting time between
+    // touchstart and touchend.
+    this.client.loader.getActions().longPress(message, 1).perform();
+    this.menuAccessors.selectAppMenuOption('View message report');
+
+    var ReportView = require('../report/view');
+    var reportView = new ReportView(this.client);
+    reportView.accessors.waitToAppear();
+    return reportView;
+  },
+
+  clearMessage: function() {
+    var messageInput = this.composerAccessors.messageInput;
+    messageInput.tap();
+    while (messageInput.text() !== '') {
+      messageInput.sendKeys(this.KEYS.backspace);
+    }
+  },
+
+  typeMessage: function(message) {
+    this.composerAccessors.messageInput.sendKeys(message);
+  },
+
+  addAttachment: function() {
+    return this.composerAccessors.addAttachment();
+  },
+
+  typeSubject: function(subject) {
+    this.composerAccessors.subjectInput.sendKeys(subject);
+  },
+
+  showOptions: function() {
+    this.accessors.optionsButton.tap();
+  },
+
+  showSubject: function() {
+    this.showOptions();
+    this.menuAccessors.selectAppMenuOption('Add subject');
+    this.client.helper.waitForElement(this.composerAccessors.subjectInput);
+  },
+
+  hideSubject: function() {
+    this.showOptions();
+    this.menuAccessors.selectAppMenuOption('Remove subject');
+    this.client.helper.waitForElementToDisappear(
+      this.composerAccessors.subjectInput
+    );
+  },
+
+  isSubjectVisible: function() {
+    var subjectInput = this.composerAccessors.subjectInput;
+    return subjectInput && this.composerAccessors.subjectInput.displayed();
+  },
+
+  backToInbox: function() {
```

Comment:
We didn't figure out better way with Johan for now for the cases when back can lead to different views - so for now it's going to be different back methods with some context in their names.